### PR TITLE
Use STBVorbis to decode ogg data for Sound

### PR DESCRIPTION
**Pull request by Hangman**
I replaced the jorbis ogg decoder by STBVorbis for Sounds (literally the Sound class, not Music) in the lwjgl3 backend. This drastically reduces the loading time, usually something between 40% and 60%, depending on the file.

"Fixes" [https://github.com/libgdx/libgdx/issues/6635](https://github.com/libgdx/libgdx/issues/6635) I guess?

Since this change will affect almost all users and I don't want to take this lightly, I'd like to mention the limitations of the decoder, that ... I don't understand ... but maybe someone else:

- floor 0 not supported (used in old ogg vorbis files pre-2004)
- lossless sample-truncation at beginning ignored
- cannot concatenate multiple vorbis streams
- sample positions are 32-bit, limiting seekable 192Khz files to around 6 hours (Ogg supports 64-bit)

I only tested on Windows 11.

### CODE REVIEW

**Comment by BambooBandit** [comment]
I gave this change a try and sheesh its a lot faster. Haven't noticed any issues either. Tried it on win 10 LWJGL3

**Comment by Berstanio** [comment]
Seems to work fine on MacOS too and it is really faster.
Is there a reason to only apply this to Sounds? I think it might be good to have it consistent for Sounds and Music.

**Comment by Hangman** [comment]
I agree, consistency would be nice. However, I'm not entirely convinced that this alone justifies a rewrite of `OggInputStream`.

A few thoughts on this that speak against it for me:
Compared to this rather simple change, an implementation for Music would be much more complex. Because of files packed in jars, we can't let STVorbis open/read on its own, so we'd have to buffer both the encoded and the decoded data. Performance-wise I'm unsure if it is for the worse or the better with such small buffer sizes.

**tl;dr**
If we exclude the performance aspect ([#6509 ](https://github.com/libgdx/libgdx/pull/6509) should help), I am left with:
\+ consistency
\- replace a battle-tested implementation with a new one
\- it takes more time to write, test, and review (debatable, but I'd rather spend the manpower on other PRs like the one mentioned)

**Comment by Berstanio** [comment]
Yeah, I looked a bit deeper into it and it certainly is more complicated to implement.
So I agree, this PR is good on its own in my opinion.

The performance difference would be still interesting to investigate at some point though.

**Review by Berstanio — APPROVED** [review]

**Review by crykn — APPROVED** [review]
